microcosm_eventsource/stores/rollup.py

```diff
@@ -0,0 +1,156 @@
+"""
+Rolled up event store.
+
+"""
+from sqlalchemy import func
+from sqlalchemy.orm import aliased
+
+from microcosm_postgres.context import SessionContext
+
+
+class RolledUpEventStore:
+
+    def __init__(self, container_store, event_store):
+        self.container_store = container_store
+        self.event_store = event_store
+
+    @property
+    def container_type(self):
+        return self.container_store.model_class
+
+    @property
+    def event_type(self):
+        return self.event_store.model_class
+
+    def retrieve(self, identifier):
+        # XXX TODO
+        raise NotImplementedError
+
+    def count(self, **kwargs):
```

Comment:
There's a lot of detail starting here. I haven't tested nearly enough, but getting here took a lot longer than you might think; the combination of a) making the window query join on the (filter) container query, b) extending to new aggregates, and c) properly extracting both the event and container as SQLAlchemy entity was non-trivial.